### ISSUE TITLE
Convert to React

### ISSUE THREAD

**Issue by seriema** [episerver/musicfestival-vue-template#9]
Create a `musicfestival-react-template` template site based on this repo but written in React. With a suitable conversion of the CSS.

**Comment by seriema** [comment]
If anyone wants to contribute, you can create a branch with the conversion and submit a PR where you'll sign our CLA. From there we can create a new repo for the react version.

/JP, Team Lead for CMS UI @ Episerver

**Comment by mattvl** [comment]
I'm working on porting this project to react, but I don't have a working version yet.
Since I've read that @jtonvang is also working on it, I'd like to ask him how he solved the part where the url query params are gotten from the _{ rootState }_ object in the _UPDATE_MODEL_BY_CONTENT_LINK_ action in _epiDataModel.js_. I am not able to find a way to make it work like that in react.

Thanks in advance for any suggestion.

**Comment by jtonvang** [comment]
@mattvl If you use Redux and React Router, you could update the Redux state with the current params in your event that handles the route change. Then you could get the params from Redux whenever you need them.

**Comment by seriema** [comment]
Hi @mattvl and @jtonvang! Have either of you gotten anywhere and have something online that we can look at?

**Comment by jtonvang** [comment]
Hi @seriema, sorry for not getting back to you sooner, I'm just back at work after paternity leave and vacation. Unfortunately, I didn't find the time for this project during summer, but I'll see if I can get some time for it in the coming weeks depending on my project allocation.

**Comment by seriema** [comment]
Welcome back @jtonvang! Don't worry, no one with kids really expected you to find the time 🤣 If you would guesstimate, how much time would it take you to convert the whole repo?

**Comment by CharbelIlias** [comment]
Hello, @seriema  and @jtonvang, we are thinking about doing this. Have you guys come any closer to doing this?

**Comment by seriema** [comment]
Hi @CharbelIlias! We haven't done anything on it, but I did find that someone has done the work of converting the actual components to React: https://github.com/jacobhytter/react-epi
Known issues with that site:
* Styling
  * Backgrounds missing
  * Overall styles missing or wrong (footer, buttons, artist lists, etc)
* OPE
  * Content Area
    * Overlays not showing per block
    * Changing view size (automatic -> narrow) doesn't update
  * Navigating on the site (clicking links etc) does not work
  * BuyTicket button does not show a dialog

@jtonvang did convert everything but the actual components I think. If he's ok with sharing that part you'd have a great start!

**Comment by CharbelIlias** [comment]
Thank you for answering, @seriema ! 

Exciting! We cloned the repo but cant login to the epi-backend. Do you know what login to use? 

@jtonvang if you would be willing to share your progress we would appreciate that, and we will also upload our solution!..

**Comment by seriema** [comment]
See the readme @CharbelIlias 😁

**Comment by CharbelIlias** [comment]
Yes, we had a collegue who couldnt login with the readme login, but we made it work. :) Thx again.

**Comment by jtonvang** [comment]
@CharbelIlias Sure, I'll send you our infrastructure to the e-mail in your profile. It basically follows the same pattern as the Vue-MusicFestical site, but ported to React/Redux. Let me know if you have any questions :)

**Comment by jtonvang** [comment]
@CharbelIlias I got a message that your gmail blocked the attachment - do you have another e-mail address I could sent it to?

**Comment by CharbelIlias** [comment]
@jtonvang yes you can try charbel.ilias@afaforsakring.se  

Thank you very much, and as soon as we have a working template, we will contribute back. :)

**Comment by seriema** [comment]
@CharbelIlias Feel free to fork this repo and work on it there! https://help.github.com/en/github/getting-started-with-github/fork-a-repo
It would allow us and others to help out, and make it easier to contribute.

**Comment by CharbelIlias** [comment]
@seriema we are building with epi-foundation cms as a base api, and frontend react, so we took inspiration from this repo. I'll fork it and start porting over the react part of it as soon as we have ironed everything out. :)

**Comment by seriema** [comment]
@CharbelIlias Hi! How did it go? Can we help you with something?

**Comment by KDunin** [comment]
Hi guys!
So I've forked this example and replaced vue with react some time ago. There are some improvements needed, but it's working ok. Wanted to share finished example, but I had no time to work on it since December, so I decide to share what I've got.
[react-epi-editable](https://github.com/KDunin/react-epi-editable)

**Comment by seriema** [comment]
@KDunin Thank you!
@CharbelIlias How did it go with your port? Have a look at KDunin's work and perhaps you can finish it?
